service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -35,4 +37,46 @@ public void disbandTeam(JWTData userData, int teamId) {
     team.setDeletedAt(new Timestamp(System.currentTimeMillis()));
     team.store();
   }
+
+  @Override
+  public void createTeam(JWTData userData, CreateTeamRequest request) {
+    TeamsRecord team = db.newRecord(TEAMS);
+    team.setBio(request.getBio());
+    team.setTeamName(request.getName());
+    team.store();
+    UsersTeamsRecord usersTeam = db.newRecord(USERS_TEAMS);
+    usersTeam.setTeamId(team.getId());
+    usersTeam.setTeamRole(TeamRole.LEADER);
+    usersTeam.setUserId(userData.getUserId());
+    usersTeam.store();
+  }
+
+  @Override
+  public TeamDataResponse getTeam(JWTData userData, GetTeamRequest getTeamRequest) {
+    TeamsRecord team =
+        db.selectFrom(TEAMS).where(TEAMS.ID.eq(getTeamRequest.getTeamId())).fetchOne();
+    if (team == null) {
+      throw new TeamDoesNotExistException(getTeamRequest.getTeamId());
+    }
+
+    return new TeamDataResponse(
+        team.getId(),
+        team.getTeamName(),
+        team.getBio(),
+        team.getFinished(),
+        team.getCreatedAt(),
+        team.getDeletedAt());
+  }
+
+  @Override
+  public void addGoal(JWTData userData, AddGoalRequest addGoalRequest) {
+    // TODO if user is not leader. thror error and send 401 unauthorized
+    GoalsRecord goal = db.newRecord(GOALS);
+    goal.setGoal(addGoalRequest.getGoal());
+    goal.getStartAt(addGoalRequest.getCompleteBy());
+    goal.getCompleteBy(addGoalRequest.getCompleteBy());
```

Comment:
Don't forget to call `goal.store()` otherwise jOOQ is never going to add this record to the database